app/create_app.py

```diff
@@ -13,6 +19,23 @@
 from app.db.models import Fund  # noqa:F401
 from app.db.models import Round  # noqa:F401
 
+PUBLIC_ROUTES = [
+    "static",
+    "build_fund_bp.healthcheck",
+    "build_fund_bp.index",
+    "build_fund_bp.login",
+]
+
+
+def protect_private_routes(flask_app: Flask) -> Flask:
+    for endpoint, view_func in flask_app.view_functions.items():
+        if endpoint in PUBLIC_ROUTES:
+            continue
+        flask_app.view_functions[endpoint] = login_required(
+            check_internal_user(view_func), return_app=SupportedApp.FUND_APPLICATION_BUILDER
+        )
+    return flask_app
```

Comment:
This does work I suppose, but it's maybe a bit unconventional.

I think adding a `before_request` decorator to the entire flask app is probably more like what I normally see?

Something like this inside create_app:

```suggestion
    @flask_app.before_request
    def protect_private_routes():
        print(request.endpoint)
        if request.endpoint in PUBLIC_ROUTES:
            return

        return login_required(lambda: None, return_app=SupportedApp.FUND_APPLICATION_BUILDER)()
```

Reply:
Discussed over Slack, will stick with current implementation. Benefits of current implementation are that routes are protected once at app startup rather than with a per-request check, and that we avoid the slightly obtuse `lambda: None` syntax. Sam highlighted though that this implementation is slightly non-standard, as he would typically expect to find middleware set either on views directly, or on blueprints or the app with `before_request` and `after_request`